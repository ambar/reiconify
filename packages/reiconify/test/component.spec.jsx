const React = require('react')
const renderer = require('react-test-renderer')
const fsp = require('fs/promises')
const path = require('path')
const transformFiles = require('../lib/transformFiles')

describe('component', () => {
  const fixtureDir = path.resolve(path.resolve(__dirname, 'fixtures/component'))
  const esDir = path.resolve(fixtureDir, 'src')
  let Icons

  beforeAll(async () => {
    await transformFiles({
      cwd: fixtureDir,
      inputs: 'icons/*.svg',
      es: true,
      esDir,
    })
    Icons = require(esDir)
  })

  afterAll(async () => {
    await fsp.rm(esDir, {recursive: true, force: true})
  })

  it('renders `size` prop', () => {
    const tree = renderer.create(<Icons.Check size={20} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders `className` prop', () => {
    const tree = renderer.create(<Icons.Check className="myIcon" />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders `text` prop', () => {
    const tree = renderer.create(<Icons.Check text />).toJSON()
    expect(tree).toMatchSnapshot()

    const tree2 = renderer.create(<Icons.Check text={false} />).toJSON()
    expect(tree2).toMatchSnapshot()
  })

  it('renders `center` prop', () => {
    const tree = renderer.create(<Icons.Check center />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders `text` and `center` prop', () => {
    const tree = renderer.create(<Icons.Check text center />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders `center` and `style` prop', () => {
    const tree = renderer
      .create(<Icons.Check center style={{color: 'red'}} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
