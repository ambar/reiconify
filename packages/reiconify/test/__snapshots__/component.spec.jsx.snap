// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component renders \`center\` and \`style\` prop 1`] = `
<span
  style={
    {
      "alignItems": "center",
      "display": "inline-flex",
    }
  }
>
  ​
  <svg
    className="MyIcon MyIcon--Check"
    fill="currentColor"
    height="24"
    style={
      {
        "color": "red",
      }
    }
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
    />
  </svg>
</span>
`;

exports[`component renders \`center\` prop 1`] = `
<span
  style={
    {
      "alignItems": "center",
      "display": "inline-flex",
    }
  }
>
  ​
  <svg
    className="MyIcon MyIcon--Check"
    fill="currentColor"
    height="24"
    viewBox="0 0 24 24"
    width="24"
  >
    <path
      d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
    />
  </svg>
</span>
`;

exports[`component renders \`className\` prop 1`] = `
<svg
  className="MyIcon MyIcon--Check myIcon"
  fill="currentColor"
  height="24"
  viewBox="0 0 24 24"
  width="24"
>
  <path
    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
  />
</svg>
`;

exports[`component renders \`size\` prop 1`] = `
<svg
  className="MyIcon MyIcon--Check"
  fill="currentColor"
  height={20}
  viewBox="0 0 24 24"
  width={20}
>
  <path
    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
  />
</svg>
`;

exports[`component renders \`text\` and \`center\` prop 1`] = `
<span
  style={
    {
      "alignItems": "center",
      "display": "inline-flex",
    }
  }
>
  ​
  <svg
    className="MyIcon MyIcon--Check"
    fill="currentColor"
    height="1.2em"
    viewBox="0 0 24 24"
    width="1.2em"
  >
    <path
      d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
    />
  </svg>
</span>
`;

exports[`component renders \`text\` prop 1`] = `
<svg
  className="MyIcon MyIcon--Check"
  fill="currentColor"
  height="1.2em"
  viewBox="0 0 24 24"
  width="1.2em"
>
  <path
    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
  />
</svg>
`;

exports[`component renders \`text\` prop 2`] = `
<svg
  className="MyIcon MyIcon--Check"
  fill="currentColor"
  height="24"
  viewBox="0 0 24 24"
  width="24"
>
  <path
    d="M9 16.17L4.83 12l-1.42 1.41L9 19 21 7l-1.41-1.41z"
  />
</svg>
`;
